src/marshmallow/fields.py

```diff
@@ -1461,7 +1480,9 @@ class TimeDelta(Field):
         "format": "{input!r} cannot be formatted as a timedelta.",
     }
 
-    def __init__(self, precision: str = SECONDS, **kwargs):
+    def __init__(
+        self, precision: str = SECONDS, serde_type: type[int | float] = int, **kwargs
```

Comment:
I don't really like the name. I suppose you mean serialization/deserialization type. Although it is involved in both serialization and deserialization, to me the name implies the destination type, so it is the serialization type while the deserialization type is `dt.timedelta`.

Therefore, I'd call it `serialization_type` or `ser_type`. Those doesn't sound so nice, so I'm open to suggestions.

Reply:
Went with `serialization_type`... Maybe `primitive_type` could be used but I feel the former describes it better.